source/index.ts

```diff
@@ -100,6 +102,7 @@ class APIClient {
     async uploadExisting(
         readStream: ReadStream | ReadableStream,
         token: string | Promise<string> = this.fetchToken(),
+        waitSuccessRetries = 0,
```

Comment:
People generally think in timeouts, not amount of retries, so let's change this to a time-based option. But let's figure out the required API for this first, it's not clear where this polling belongs.